bool start
bool cancel
bool reset
int32 distance